femiarkh/shop-react-redux-cloudfront-be
# Task 7

### 1. [Task](https://github.com/EPAM-JS-Competency-center/cloud-development-course-initial/blob/main/task7-lambda%2Bcognito-authorization/task.md)
### 2. Deadline: 17.10.2021 / Done: 15.10.2021
### 3. Self-evaluation: 6/6
- 1 - authorization-service is added to the repo, has correct basicAuthorizer lambda and correct serverless.yaml file
- 3 - import-service serverless.yaml file has authorizer configuration for the importProductsFile lambda. Request to the importProductsFile lambda should work only with correct authorization_token being decoded and checked by basicAuthorizer lambda. Response should be in 403 HTTP status if access is denied for this user (invalid authorization_token) and in 401 HTTP status if Authorization header is not provided.
- 5 - update client application to send Authorization: Basic authorization_token header on import. Client should get authorization_token value from browser localStorage https://developer.mozilla.org/ru/docs/Web/API/Window/localStorage authorization_token = localStorage.getItem('authorization_token')
- +1 - Client application should display alerts for the responses in 401 and 403 HTTP statuses. This behavior should be added to the nodejs-aws-fe-main/src/index.tsx file

### 4. Links:
- [Frontend PR](https://github.com/femiarkh/shop-react-redux-cloudfront/pull/5)
- [Deployed shop app](https://d2uxsq2yxa1sxw.cloudfront.net/)
Your test csv file may be something like:
```
title,description,price,count,image
bikeone,description,50,100,
biketwo,description,150,100,
```

### 5. How to test:
1. Open deployed front app, go to Manage products page and try to upload some `csv` file with new products (see example above).
2. Note `Unauthorized` alert and 401 error in console.
3. Open devtools --> Application tab --> Local storage --> https://d2uxsq2yxa1sxw.cloudfront.net/. Add a record there:
Key = `authorization_token`
Value = some random string
![image](https://user-images.githubusercontent.com/55946100/137538480-80946b10-92a9-4870-912b-6417c8412623.png)
4. Reload the page and try to upload some `csv` file ones again.
5. Note `User is not authorized to access this resource with an explicit deny` alert and 403 error in console. 
6. Change value in local storage to `ZmVtaWFya2g6VEVTVF9QQVNTV09SRA==` (this is `femiarkh:TEST_PASSWORD` encoded in Base64). 
![image](https://user-images.githubusercontent.com/55946100/137538568-5940ec4b-601a-412c-a651-d401060a804b.png)
7. Reload the page and try to upload some `csv` file ones again.
8. There should be no error, and new products should appear in the catalog after some time.
